src/auth/apiheader/core/qgsauthapiheadermethod.cpp

```diff
@@ -0,0 +1,172 @@
+/***************************************************************************
+    qgsauthapiheadermethod.cpp
+    --------------------------
+    begin                : October 2021
+    copyright            : (C) 2021 by Tom Cummins
+    author               : Tom Cummins
+    email                : tom cumminsc9 at googlemail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsauthapiheadermethod.h"
+
+#include "qgsauthmanager.h"
+#include "qgslogger.h"
+#include "qgsapplication.h"
+#include "qgsmessagelog.h"
+
+#ifdef HAVE_GUI
+#include "qgsauthapiheaderedit.h"
+#endif
+
+#include <QNetworkProxy>
+#include <QMutexLocker>
+#include <QUuid>
+
+const QString QgsAuthApiHeaderMethod::AUTH_METHOD_KEY = QStringLiteral( "APIHeader" );
+const QString QgsAuthApiHeaderMethod::AUTH_METHOD_DESCRIPTION = QStringLiteral( "API Header" );
+const QString QgsAuthApiHeaderMethod::AUTH_METHOD_DISPLAY_DESCRIPTION = tr( "API Header" );
+
+QMap<QString, QgsAuthMethodConfig> QgsAuthApiHeaderMethod::sAuthConfigCache = QMap<QString, QgsAuthMethodConfig>();
+
+
+QgsAuthApiHeaderMethod::QgsAuthApiHeaderMethod()
+{
+  setVersion( 2 );
+  setExpansions( QgsAuthMethod::NetworkRequest );
+  setDataProviders( QStringList()
+                    << QStringLiteral( "ows" )
+                    << QStringLiteral( "wfs" )  // convert to lowercase
+                    << QStringLiteral( "wcs" )
+                    << QStringLiteral( "wms" ) );
+}
+
+QString QgsAuthApiHeaderMethod::key() const
+{
+  return AUTH_METHOD_KEY;
+}
+
+QString QgsAuthApiHeaderMethod::description() const
+{
+  return AUTH_METHOD_DESCRIPTION;
+}
+
+QString QgsAuthApiHeaderMethod::displayDescription() const
+{
+  return AUTH_METHOD_DISPLAY_DESCRIPTION;
+}
+
+bool QgsAuthApiHeaderMethod::updateNetworkRequest( QNetworkRequest &request, const QString &authcfg,
+    const QString &dataprovider )
+{
+  Q_UNUSED( dataprovider )
+  const QgsAuthMethodConfig mconfig = getMethodConfig( authcfg );
```

Comment:
```suggestion
  const QgsAuthMethodConfig config = getMethodConfig( authcfg );
```

(The "m" prefix is used in qgis to indicate member variables)